Formula/w/whatweb.rb

```diff
@@ -0,0 +1,38 @@
+class Whatweb < Formula
+  desc "Web scanner. Identify the technology stack that powers a website"
+  homepage "https://morningstarsecurity.com/research/whatweb"
+  url "https://github.com/urbanadventurer/WhatWeb/archive/refs/tags/v0.5.5.tar.gz"
+  sha256 "96dedb6a377184fb8f5fd3f2a81c26ff8c92c4dc1503ce409793a1e7ab23695d"
+  license "GPL-2.0-only"
+  head "https://github.com/urbanadventurer/WhatWeb.git", branch: "master"
+
+  depends_on "rbenv"
+  depends_on "ruby-build"
+  uses_from_macos "ruby" => :build
+
+  # newer ruby version is needed for older macos
+  on_monterey :or_older do
+    depends_on "ruby" => :build
+  end
+
+  def install
+    ENV["GEM_HOME"] = buildpath/"gem_home"
+    system "gem", "install", "bundler"
+    ENV.prepend_path "PATH", buildpath/"gem_home/bin"
+    system "bundle", "install"
+    system "make", "install",
+      "prefix=#{prefix}",
+      "-e", "BINPATH=#{bin}",
+      "-e", "DOCPATH=#{doc}",
+      "-e", "LIBPATH=#{lib}",
+      "-e", "MANPATH=#{man}"
+    system "gem", "install", "bson"
+    system "gem", "install", "bson_ext"
+    system "gem", "install", "mongo"
+    system "gem", "install", "rchardet"
+  end
+
+  test do
+    assert_equal "WhatWeb version #{version} ( https://www.morningstarsecurity.com/research/whatweb/ )", shell_output("#{bin}/whatweb --version").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
The test now includes a simple test that uses `google.com` as a target. I then use `assert_match` to test the application is working as expected.

I kept the version test in addition